x/emissions/keeper/msgserver/msg_server_stake.go

```diff
@@ -111,9 +111,14 @@ func (ms msgServer) ConfirmRemoveStake(ctx context.Context, msg *types.MsgConfir
 	if err != nil {
 		return nil, err
 	}
-	if stakeRemoval.BlockRemovalStarted+moduleParams.RemoveStakeDelayWindow >= currentBlock {
+	// if the delay window has not yet passed, its too early
+	if currentBlock < stakeRemoval.BlockRemovalStarted+moduleParams.RemoveStakeDelayWindow {
```

Comment:
if you wanted you could calculate this sum once (and re-use in the next statement), it would also look cleaner